src/solvers/flattening/flatten_byte_operators.cpp

```diff
@@ -242,12 +242,12 @@ exprt flatten_byte_extract(
 
         byte_extract_exprt tmp(unpacked);
         tmp.type()=subtype;
-        tmp.offset()=simplify_expr(new_offset, ns);
+        tmp.offset()=new_offset;
```

Comment:
Also wondering about the right place for this -- perhaps where flatten_byte_* is used?

Reply:
I am also kind of tempted to do the flattening in the simplifier.

Reply:
I do agree that this should be moved to the simplifier (also note that it only has dependencies within util/), but may I suggest that this happens as a separate PR? Otherwise this PR will become huge.

Reply:
yes, separate PR

Reply:
Thanks, I've put a note on my TODO list. I'll take care of it later on.